### PR TITLE
Reordered metadata methods

### PR DESCRIPTION
**Pull request by charlottekostelic**
Changed default `status_forcelist` in `_session.py`
Added more tests to `_session.py`
Changed method names and reordered `MetadataSession`

### CODE REVIEW

File: bookops_worldcat/metadata_api.py [hunk 1]
```diff
@@ -1728,43 +1642,61 @@ def search_my_holdings(
...
         Returns:
             `requests.Response` object
         """
-        if not q:
-            raise TypeError("Argument 'q' is requried to construct query.")
+        if not any([oclcNumber, isbn, issn]):
+            raise TypeError(
+                "Missing required argument. "
+                "One of the following args are required: oclcNumber, issn, isbn"
+            )
```
Comment (klinga) [inline bookops_worldcat/metadata_api.py:1684]: This at least should keep users sane....

File: bookops_worldcat/metadata_api.py [hunk 2]
```diff
@@ -1775,63 +1707,97 @@ def search_my_local_bibs(
...
+                                        used for signal event handling, see more at:
+                                        https://requests.readthedocs.io/en/master/user/advanced/#event-hooks
         Returns:
             `requests.Response` object
         """
+        if not any([oclcNumber, isbn, issn]):
+            raise TypeError(
+                "Missing required argument. "
+                "One of the following args are required: oclcNumber, issn, isbn"
+            )
```
Comment (klinga) [inline bookops_worldcat/metadata_api.py:1777]: 👍

File: bookops_worldcat/metadata_api.py [hunk 3]
```diff
@@ -1843,51 +1809,85 @@ def search_shared_print_lhr(
...
+            lon:                        limit to longitute, example: -122.22702
+            distance:                   distance from latitude and longitude
+            unit:                       unit of distance param; options:
+                                        'M' (miles) or 'K' (kilometers)
+            hooks:                      Requests library hook system that can be
+                                        used for signal event handling, see more at:
+                                        https://requests.readthedocs.io/en/master/user/advanced/#event-hooks
         Returns:
-            `requests.Response` instance
+            `requests.Response` object
```
Comment (klinga) [inline bookops_worldcat/metadata_api.py:1866]: instance?

**Review by klinga — COMMENTED** [review]
That was a lot of work! Impressed! 👏 
There are some issues I would like you to look at and consider before merging.

File: bookops_worldcat/metadata_api.py [hunk 4]
```diff
@@ -82,49 +82,34 @@ def _split_into_legal_volume(
         for i in range(0, len(oclc_numbers), n):
             yield ",".join(oclc_numbers[i : i + n])  # noqa: E203
```
[the remark this answers is not in the document]
Comment (charlottekostelic) [reply bookops_worldcat/metadata_api.py:83]: This method is still used in `holdings_get_current` but I could change that method to specify that users should supply no more than 10 oclc numbers

File: bookops_worldcat/metadata_api.py [hunk 5]
```diff
@@ -166,38 +166,51 @@ def _url_search_lbd_control_number(self, controlNumber: Union[str, int]) -> str:
...
                                     used for signal event handling, see more at:
                                     https://requests.readthedocs.io/en/master/user/advanced/#event-hooks
         Returns:
             `requests.Response` instance
         """
-        oclcNumber = verify_oclc_number(oclcNumber)
+        if not record:
+            raise TypeError("Argument 'record' is missing.")
+        if not recordFormat:
+            raise TypeError("Argument 'recordFormat' is missing.")
```
[the remark this answers is not in the document]
Comment (charlottekostelic) [reply bookops_worldcat/metadata_api.py:196]: I'll remove these checks. 

Maybe an enhancement could be to check that the record matches the specified record format and raise an error if they don't match? I can open an issue to add it as a future enhancement.

File: bookops_worldcat/metadata_api.py [hunk 6]
```diff
@@ -256,241 +299,199 @@ def get_institution_holdings(
...
+            hooks:                  Requests library hook system that can be
+                                    used for signal event handling, see more at:
+                                    https://requests.readthedocs.io/en/master/user/advanced/#event-hooks
+        Returns:
+            `requests.Response` instance
         """
         oclcNumber = verify_oclc_number(oclcNumber)
 
-        url = self._url_search_brief_bibs_other_editions(oclcNumber)
+        url = self._url_search_brief_bibs_oclc_number(oclcNumber)
```
[the remark this answers is not in the document]
Comment (charlottekostelic) [reply bookops_worldcat/metadata_api.py:482]: Just read your next comment. I'll leave this as `brief_bibs_get`

File: bookops_worldcat/metadata_api.py [hunk 7]
```diff
@@ -1431,265 +1337,270 @@ def holding_set_on_record(
...
+        req = Request("GET", url, params=payload, headers=header, hooks=hooks)
         prepared_request = self.prepare_request(req)
 
         # send request
         query = Query(self, prepared_request, timeout=self.timeout)
 
         return query.response
 
-    def search_my_holdings(
+    def local_holdings_search_shared_print(
```
[the remark this answers is not in the document]
Comment (charlottekostelic) [reply bookops_worldcat/metadata_api.py:1577]: The quality of OCLC's documentation gets worse the further down the page one gets.

**Comment by charlottekostelic** [comment]
@klinga, I made all of the requested edits and went through the args for each endpoint to clarify which are required and which are optional. In most cases we only have default values for args that OCLC has provided a default value for. The only instances in which this is not the case is with `preferredLanguage` and `inCatalogLanguage` which have 'eng' as the default value. 

There are some args which are technically optional but one of a few options must be included in a query (eg. `summary_holdings_get` must contain either `oclcNumber`, `isbn`, or `issn`). I added this and similar details to the doc strings to hopefully make it a bit clearer to users but if you have a suggestion about a better way to document this, let me know. The error messages from the API are mostly helpful but some of them are misleading and suggest using args that aren't actually available for a particular endpoint.

**Pull request merged**